platform/src/share/mod.rs

```diff
@@ -0,0 +1,21 @@
+//! `share` contains tools for safely sharing objects (such as buffers and
+//! upcalls) with the Tock kernel.
+
+mod handle;
+mod tuple_impls;
+
+pub use handle::{Handle, SplittableHandle};
+
+/// Creates a scope in which objects may safely be shared with the kernel.
+pub fn scope<L: List, Output, F: FnOnce(Handle<L>) -> Output>(fcn: F) -> Output {
+    let list = Default::default();
+    fcn(unsafe { Handle::new(&list) })
```

Comment:
nit: add safety comment that list isn't getting forgotten or leaked to this is safe

Reply:
Added, good catch!